cuda/cuda_nemolite2d.cu

```diff
@@ -488,63 +622,96 @@ k_momentum()
 }
 
 __global__ void
-k_boundary_conditions()
-{
-  // TODO:
-}
+k_boundary_conditions(wp_t rtime,
+                      const FortranArray2D<wp_t, 0, 1>& sshn_u,
+                      const FortranArray2D<wp_t, 1, 0>& sshn_v,
 
-__global__ void
-k_next()
-{
-  // TODO:
-}
+                      const FortranArray2D<wp_t, 1, 1>& ssha,
 
-void
-cuda_finalise_()
-{
-  // Clean up grid constants arrays.
-  delete grid_constants.e1t;
-  delete grid_constants.e2t;
-  delete grid_constants.e1u;
-  delete grid_constants.e2u;
+                      const FortranArray2D<wp_t, 0, 1>& ua,
+                      const FortranArray2D<wp_t, 1, 0>& va,
 
-  delete grid_constants.e1f;
-  delete grid_constants.e2f;
-  delete grid_constants.e1v;
-  delete grid_constants.e2v;
+                      const FortranArray2D<wp_t, 0, 1>& ua_buffer,
+                      const FortranArray2D<wp_t, 1, 0>& va_buffer,
 
-  delete grid_constants.e12t;
-  delete grid_constants.e12u;
-  delete grid_constants.e12v;
+                      const FortranArray2D<wp_t, 0, 1>& hu,
+                      const FortranArray2D<wp_t, 1, 0>& hv,
 
-  delete grid_constants.gphiu;
-  delete grid_constants.gphiv;
-  delete grid_constants.gphif;
+                      const FortranArray2D<int, 0, 0>& pt,
 
-  delete grid_constants.xt;
-  delete grid_constants.yt;
+                      const int jpi,
+                      const int jpj)
+{
+  const wp_t amp_tide = 0.2;
+  const wp_t omega_tide = (2.0 * 3.14159) / (12.42 * 3600.0);
 
-  delete grid_constants.ht;
-  delete grid_constants.hu;
-  delete grid_constants.hv;
+  int ji = threadIdx.x * blockIdx.x + blockDim.x;
```

Comment:
Using 2D kernel indexing, but currently launched as a 1D kernel. See the comment above.